src/main/java/com/importer/importer/DataSourceConfig.java

```diff
@@ -19,18 +18,35 @@ public class DataSourceConfig {
 
     @Value("${spring.datasource.username}")
     private String dataBaseUsername;
+    
+    @Value("${spring.datasource.password}")
+    private String dataBasePassword;
 
     @Autowired
-    private Environment environment;
+    private KMSUtil kmsUtil;
 
     @Bean
     public DataSource dataSource() throws IOException {
-        String dbPassword = environment.getProperty("spring.datasource.password");
+       // String dbPassword = environment.getProperty("spring.datasource.password");
+        boolean isEncrypted = checkDbPasswordIsEncrypted(dataBasePassword);
+        if(isEncrypted){
+          dataBasePassword = decryptEncryptedPassword(dataBasePassword);
+        }
         DriverManagerDataSource dataSource = new DriverManagerDataSource();
         dataSource.setDriverClassName("org.postgresql.Driver");
         dataSource.setUrl(dataBaseUrl);
         dataSource.setUsername(dataBaseUsername);
-        dataSource.setPassword(dbPassword);
+        dataSource.setPassword(dataBasePassword);
         return dataSource;
     }
+
+    private boolean checkDbPasswordIsEncrypted(String dataBasePassword) {
+        return dataBasePassword.startsWith("ENC(") & dataBasePassword.endsWith(")");
+    }
+
+    private String decryptEncryptedPassword(String dataBasePassword) {
```

Comment:
The names of the method and variable are not intuitive. Please think of a big picture. 
This operation will be performed on not just the db password but other encrypted props too.